source/plugins/ruby/out_named_pipe.rb

```diff
@@ -0,0 +1,105 @@
+require 'fluent/plugin/output'
+
+module Fluent::Plugin
+  class NamedPipeOutput < Output
+    Fluent::Plugin.register_output('named_pipe', self)
+
+    helpers :formatter
+
+    config_param :datatype, :string
+
+    unless method_defined?(:log)
+      define_method(:log) { $log }
+    end
+
+    def initialize
+        super
+        require_relative "extension_utils"
+    
+    end
+
+    def configure(conf)
+      super
+
+      @formatter = formatter_create(usage: 'msgpack_formatter', type: 'msgpack' )
+    end
+
+    def getNamedPipeFromExtension()
+        @pipe_name = ""
```

Comment:
move variable to top as this instance variable. refer to other plugins for examples.